salt/modules/virt.py

```diff
@@ -1168,22 +1168,55 @@ def _handle_remote_boot_params(orig_boot):
     """
     saltinst_dir = None
     new_boot = orig_boot.copy()
+    keys = orig_boot.keys()
 
     try:
-        for key in ['kernel', 'initrd']:
-            if check_remote(orig_boot.get(key)):
-                if saltinst_dir is None:
-                    os.makedirs(CACHE_DIR)
-                    saltinst_dir = CACHE_DIR
+        if boot_param_is_valid(keys):
+            if 'loader' in keys:
+                new_boot['nvram'] = handle_loader_tag(orig_boot, name)
+            for key in keys:
+                if check_remote(orig_boot.get(key)):
+                    if saltinst_dir is None:
+                        os.makedirs(CACHE_DIR)
+                        saltinst_dir = CACHE_DIR
+                        new_boot[key] = download_remote(orig_boot.get(key),
+                                                        saltinst_dir)
+            return new_boot
 
-                new_boot[key] = download_remote(orig_boot.get(key),
-                                                saltinst_dir)
-
-        return new_boot
     except Exception as err:  # pylint: disable=broad-except
         raise err
 
 
+def handle_loader_tag(orig_boot, name):
+    """
+    check if the path for storing VARS file exists, if not create for it.
+    rename the file and copy to the corresponding location.
+    :param name: the name of the domain
+    :param orig_boot: The original boot parameters passed to the init or update
+    :return:  string that contains the path of variable store file under qemu
+    """
+    qemu_dir = '/var/lib/libvirt/qemu/nvram'
+    if not os.path.exists(qemu_dir):
+        os.makedirs(qemu_dir)
+    dirname = os.path.dirname(orig_boot.get('loader'))
+    src = '{}/OVMF_VARS.fd'.format(dirname)
```

Comment:
in some distros like openSUSE the var templates files aren't named like this. Here is the content of the `qemu-ovmf-x86_64` package on openSUSE:

```
# rpm -ql qemu-ovmf-x86_64                                                                                                                                                                       
/usr/share/licenses/qemu-ovmf-x86_64
/usr/share/licenses/qemu-ovmf-x86_64/License-ovmf.txt
/usr/share/licenses/qemu-ovmf-x86_64/License.txt
/usr/share/qemu
/usr/share/qemu/firmware
/usr/share/qemu/firmware/50-ovmf-x86_64-secure-ms.json
/usr/share/qemu/firmware/50-ovmf-x86_64-secure-opensuse.json
/usr/share/qemu/firmware/50-ovmf-x86_64-secure-suse.json
/usr/share/qemu/firmware/50-ovmf-x86_64-secure.json
/usr/share/qemu/firmware/60-ovmf-x86_64.json
/usr/share/qemu/ovmf-x86_64-4m-code.bin
/usr/share/qemu/ovmf-x86_64-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-4m.bin
/usr/share/qemu/ovmf-x86_64-code.bin
/usr/share/qemu/ovmf-x86_64-ms-4m-code.bin
/usr/share/qemu/ovmf-x86_64-ms-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-ms-4m.bin
/usr/share/qemu/ovmf-x86_64-ms-code.bin
/usr/share/qemu/ovmf-x86_64-ms-vars.bin
/usr/share/qemu/ovmf-x86_64-ms.bin
/usr/share/qemu/ovmf-x86_64-opensuse-4m-code.bin
/usr/share/qemu/ovmf-x86_64-opensuse-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-opensuse-4m.bin
/usr/share/qemu/ovmf-x86_64-opensuse-code.bin
/usr/share/qemu/ovmf-x86_64-opensuse-vars.bin
/usr/share/qemu/ovmf-x86_64-opensuse.bin
/usr/share/qemu/ovmf-x86_64-smm-code.bin
/usr/share/qemu/ovmf-x86_64-smm-ms-code.bin
/usr/share/qemu/ovmf-x86_64-smm-ms-vars.bin
/usr/share/qemu/ovmf-x86_64-smm-ms.bin
/usr/share/qemu/ovmf-x86_64-smm-opensuse-code.bin
/usr/share/qemu/ovmf-x86_64-smm-opensuse-vars.bin
/usr/share/qemu/ovmf-x86_64-smm-opensuse.bin
/usr/share/qemu/ovmf-x86_64-smm-suse-code.bin
/usr/share/qemu/ovmf-x86_64-smm-suse-vars.bin
/usr/share/qemu/ovmf-x86_64-smm-suse.bin
/usr/share/qemu/ovmf-x86_64-smm-vars.bin
/usr/share/qemu/ovmf-x86_64-smm.bin
/usr/share/qemu/ovmf-x86_64-suse-4m-code.bin
/usr/share/qemu/ovmf-x86_64-suse-4m-vars.bin
/usr/share/qemu/ovmf-x86_64-suse-4m.bin
/usr/share/qemu/ovmf-x86_64-suse-code.bin
/usr/share/qemu/ovmf-x86_64-suse-vars.bin
/usr/share/qemu/ovmf-x86_64-suse.bin
/usr/share/qemu/ovmf-x86_64-vars.bin
/usr/share/qemu/ovmf-x86_64.bin
```

For sure we'll need to improve the heuristic to find those files from the binary one.